### ISSUE TITLE
strange fix for jsx-no-leaked-render

### ISSUE THREAD

**Issue by lukesmurray** [jsx-eslint/eslint-plugin-react#3297]
**Input File**

```tsx
import { css } from "twin.macro";

export interface sortIndicatorProps {
  direction?: "up" | "down";
  removeFromLayout?: boolean;
  className?: string;
}

export const sortIndicatorMarginLeft = "-1em";

export function SortIndicator({
  direction,
  removeFromLayout = true,
  className,
}: sortIndicatorProps) {
  return (
    <span
      className={className}
      css={[
        css`
          font-size: 0.8em;
          &:first-of-type {
            padding-right: 0.5ch;
          }
        `,
        removeFromLayout &&
          css`
            position: absolute;
            margin-left: ${sortIndicatorMarginLeft};
          `,
      ]}>
      {direction ? (direction === "down" ? "▼" : "▲") : ""}
    </span>
  );
}
```

**output file**

```tsx
import { css } from "twin.macro";

export interface sortIndicatorProps {
  direction?: "up" | "down";
  removeFromLayout?: boolean;
  className?: string;
}

export const sortIndicatorMarginLeft = "-1em";

export function SortIndicator({
  direction,
  removeFromLayout = true,
  className,
}: sortIndicatorProps) {
  return (
    <span
      className={className}
      css={[
        css`
          font-size: 0.8em;
          &:first-of-type {
            padding-right: 0.5ch;
          }
        `,
        removeFromLayout &&
          css`
            position: absolute;
            margin-left: ${sortIndicatorMarginLeft};
          `,
      ]}>
      {/* lol what is this ↓↓↓↓↓ */}
      {!!!!!!!!!!!!!!!!!!!!direction && direction === "down" && "▼"}
    </span>
  );
}
```

eslint rule setup

```js
    "react/jsx-no-leaked-render": [
      // enforce that values are coerced to booleans in jsx conditionals
      "error",
      {
        validStrategies: ["coerce"], // coerce to boolean e.g. `<div>{count && title}</div>` becomes `<div>{!!count && title}</div>`
      },
    ],
```

**Comment by lukesmurray** [comment]
The exclamation marks show up on strings and nested values.

```
                            !!!!!!!!!!!!!!!!!!column.isSorted && column.isSortedDesc && "down"
```

**Comment by ljharb** [comment]
lol this is clearly a bug. cc @Belco90

**Comment by lukesmurray** [comment]
let me know if you have trouble reproducing. to be clear there are two bugs here. the unnecessary exclamation marks and the deletion of branches from ternary expressions

**Comment by Belco90** [comment]
> the unnecessary exclamation marks and the deletion of branches from ternary expressions

No idea what's causing the former! But the latter it's easy to fix. I'm on it.

**Comment by ljharb** [comment]
@Belco90 thanks! i'd love to publish a patch release, and i'd love to include fixes for the latest no-leaked-render issues :-)

**Comment by Belco90** [comment]
@ljharb I got the two issues reported here fixed + #3292 partially fixed. Still figuring out how to prevent reporting when TS types are involved.

**Comment by Belco90** [comment]
Sorry, I forgot to mention: I can create a PR with those fixes while I figure out the last one.
